extensions/panache/hibernate-orm-panache-common/runtime/src/main/java/io/quarkus/hibernate/orm/panache/common/runtime/CommonPanacheQueryImpl.java

```diff
@@ -103,7 +110,30 @@ private CommonPanacheQueryImpl(CommonPanacheQueryImpl<?> previousQuery, String n
         }
         select.append(") ");
 
-        return new CommonPanacheQueryImpl<>(this, select.toString() + query, "select count(*) " + query);
+        return new CommonPanacheQueryImpl<>(this, select.toString() + query, "select count(*) " + query, null);
+    }
+
+    public <T> CommonPanacheQueryImpl<T> select(List<String> columns, Function<Object, T> resultMapper) {
+        if (PanacheJpaUtil.isNamedQuery(query)) {
+            throw new PanacheQueryException("Unable to perform a select on a named query");
+        }
+        if (columns == null || columns.isEmpty()) {
+            throw new PanacheQueryException("Columns must be set");
+        }
+        if (resultMapper == null) {
+            throw new PanacheQueryException("A result mapper must be set");
+        }
+        StringBuilder select = new StringBuilder("select ");
```

Comment:
A query can start by `select` so you cannot blindly add a select clause in front of it.
I think this case should be tested and leads to a `PanacheQueryException`.

Reply:
Hm, `CommonPanacheQueryImpl.project(Class<T>)` does the same. So how should we correctly detect a problematic query?

Reply:
I think both methods should throw an exception in this case.

Reply:
I've added a simple check that throws an exception if an existing query starts with `select`. I forgot to add a test though ;-). I'll add it in a minute.